sdk/keyvault/azure-security-keyvault-common/inc/azure/keyvault/common/internal/keyvault_pipeline.hpp

```diff
@@ -97,9 +97,17 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Common { n
         Azure::Core::Context const& context,
         Azure::Core::Http::HttpMethod method,
         std::function<T(Azure::Core::Http::RawResponse const& rawResponse)> factoryFn,
-        std::vector<std::string> const& path)
+        std::vector<std::string> const& path,
+        std::unique_ptr<std::map<std::string, std::string>> const& query = nullptr)
```

Comment:
Add doc comment for this param (I know it's internal, but this API already has good docs).